.github/workflows/ci.yml

```diff
@@ -6,6 +6,10 @@ on:
     - master
   pull_request:
 
+concurrency:
+  group: ${{ github.ref_name }}
+  cancel-in-progress: true
```

Comment:
What benefit do we get from this? I thought we were already running in parallel.

I also wouldn't recommend cancelling in progress jobs when one fails. Given how wide our test matrix is it's helpful to know many entries a particular problem is scoped to.

Reply:
Last I tried, cancellation wasn't automatic, but GitHub will happily show the newest run in the UI, even while old runs are still going. This ensures that doesn't happen by assigning a PR-unique identifier and cancelling the previous jobs. 

This doesn't affect cancellation on failure.

Reply:
Ah – this is about cancelling prior runs **when new commits are pushed**. I don't have strong feelings here: this addition makes sense to me 🙂